### PR TITLE
 i9 create navigation bar

### PR DESCRIPTION
**Pull request by wuryantoAji**
## Change Summary
add navbar components 

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

> any documents where I can put documentation?
> test is only recording of the navbar the recording is in this gdrive https://drive.google.com/file/d/1VYh1PsGkbngIZsPGCjn287MrGk1LgT3Y/view?usp=sharing

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**
- needs to comment the `import './assets/main.css'` in main.ts since main.css from vue makes it render incorrectly

### CODE REVIEW

File: client/src/components/Navbar.vue [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    };
+  },
+
+  methods: {
+    menuItems() {
+      return this.menu;
+    },
+  }
+};
+</script>
```
Comment (ChantelleKerr) [inline client/src/components/Navbar.vue:71]: You are using the options API here. We are using the Composition API with Typescript. 

https://vuejs.org/api/sfc-script-setup.html#basic-syntax

File: client/src/components/Navbar.vue [hunk 2]
```diff
@@ -0,0 +1,92 @@
+<template>
+  <img src="@/assets/Footer-min.jpeg" class="border"/>
+  <v-toolbar dark app color="black" class="hidden-sm-and-down mt-2">
+    <img alt="CSF Logo"  src="@/assets/CSF_Logo_WHITE.png" width="125" height="60" />
+    <v-toolbar-title class="hidden-md-and-down">Community Spirit Foundation</v-toolbar-title>
```
Comment (ChantelleKerr) [inline client/src/components/Navbar.vue:5]: We dont need the title :)

Comment (wuryantoAji) [reply client/src/components/Navbar.vue:5]: noted, will remove it

File: client/src/components/NavbarComponents.vue [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+}
+.border-container {
+  position: relative;
+}
+.v-list {
+  padding-top: 0px;
+}
+.img {
+  height: 200px;
+}
```
Comment (ChantelleKerr) [inline client/src/components/NavbarComponents.vue:100]: You can remove this

File: client/src/components/NavbarComponents.vue [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  { icon: 'mdi-star', title: 'Leaderboards', link: '/' },
+  { icon: 'mdi-login', title: 'Login', link: '/' },
+  { icon: 'mdi-pencil-box', title: 'Sign Up', link: '/signup' }
+]
+</script>
+
+<style>
+.menuIcon {
+  padding-right: 3%;
+}
```
Comment (ChantelleKerr) [inline client/src/components/NavbarComponents.vue:77]: You can remove this - vuetify does it inline. See here: https://vuetifyjs.com/en/styles/spacing/

File: client/src/components/NavbarComponents.vue [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+          <v-toolbar dark app color="black" class="hidden-md-and-up">
+            <img alt="CSF Logo" src="@/assets/CSF_Logo_WHITE.png" width="125" height="60" />
+            <v-spacer></v-spacer>
+            <v-btn icon @click="dialog = false" class="hidden-md-and-up menuIcon">
+              <v-icon>mdi-close</v-icon>
+            </v-btn>
+          </v-toolbar>
+        </v-container>
+        <v-container class="pt-0 pl-0 pb-2 pr-0 border-container">
+          <img id="app" src="@/assets/Footer-min.jpeg" class="border hidden-md-and-up" />
```
Comment (ChantelleKerr) [inline client/src/components/NavbarComponents.vue:34]: make this into v-img. See below to copy paste

File: client/src/components/NavbarComponents.vue [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+      :fullscreen="mobile"
+      transition="dialog-up-transition"
+      class="hidden-md-and-up"
+    >
+      <template v-slot:activator="{ props }">
+        <v-btn icon v-bind="props" class="hidden-md-and-up menuIcon">
+          <v-icon>mdi-menu</v-icon>
+        </v-btn>
+      </template>
+      <v-card class="overall">
```
Comment (ChantelleKerr) [inline client/src/components/NavbarComponents.vue:23]: you can set color="black" here :)

File: client/src/components/NavbarComponents.vue [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        <v-container class="pa-0 ma-0">
+          <v-toolbar dark app color="black" class="hidden-md-and-up">
+            <img alt="CSF Logo" src="@/assets/CSF_Logo_WHITE.png" width="125" height="60" />
+            <v-spacer></v-spacer>
+            <v-btn icon @click="dialog = false" class="hidden-md-and-up menuIcon">
```
Comment (ChantelleKerr) [inline client/src/components/NavbarComponents.vue:28]: menuIcon not needed

**Review by ChantelleKerr — CHANGES_REQUESTED** [review]
Nice job! It looks really good

Some small changes and vuetify tip ;)

We want to avoid css as much as possible if vuetify can do the same thing

**Comment by wuryantoAji** [comment]
noted, will fix all of it. thanks for the review Chantelle

**Review by ChantelleKerr — APPROVED** [review]
Great job! Looks amazing. Approved

**Pull request merged**